mempool/clist_mempool.go

```diff
@@ -100,6 +99,32 @@ func NewCListMempool(
 	return mp
 }
 
+func (mem *CListMempool) getCElement(txKey types.TxKey) (*clist.CElement, bool) {
+	if e, ok := mem.txsMap.Load(txKey); ok {
+		return e.(*clist.CElement), true
+	}
+	return nil, false
+}
+
+func (mem *CListMempool) getMemTx(txKey types.TxKey) *mempoolTx {
+	if e, ok := mem.getCElement(txKey); ok {
+		return e.Value.(*mempoolTx)
+	}
+	return nil
+}
+
+func (mem *CListMempool) removeAllTxs() {
+	for e := mem.txs.Front(); e != nil; e = e.Next() {
+		mem.txs.Remove(e)
+		e.DetachPrev()
```

Comment:
The contract says "and/or": "Caller must call e.DetachPrev() and/or e.DetachNext() to avoid memory leaks". I haven't actually checked the code to know how it works. In any case, it's really good that you have checked this stuff.